File: src/network/packet/impl/response/PlayerNotificationCheckResponsePacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl\response;

use pocketcloud\cloud\network\packet\ResponsePacket;
use pocketcloud\cloud\network\packet\util\PacketData;

final class PlayerNotificationCheckResponsePacket extends ResponsePacket {

    public function __construct(private readonly bool $enabled = false) {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->writeAll($this->enabled);
    }

    public function isEnabled(): bool {
        return $this->enabled;
    }

    public static function create(bool $enabled): self {
        return new self($enabled);
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/network/packet/impl/response/PlayerNotificationCheckResponsePacket.java
package de.pocketcloud.cloud.network.packet.impl.response;

import de.pocketcloud.cloud.network.packet.ResponsePacket;
import de.pocketcloud.network.packet.data.PacketData;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Getter
public final class PlayerNotificationCheckResponsePacket extends ResponsePacket {

    private boolean enabled;

    public PlayerNotificationCheckResponsePacket(boolean enabled) {
        this.enabled = enabled;
    }

    @Override
    public void encodePayload(PacketData packetData) {
        packetData.writeAll(enabled);
    }

    public static PlayerNotificationCheckResponsePacket create(boolean enabled) {
        return new PlayerNotificationCheckResponsePacket(enabled);
    }
}
